Parsely/parsely-android
# Coroutines: add event flow

### Description

This PR introduces Kotlin Coroutines for "add an event" flow. It covers
- adding an event to the in-memory buffer
- storing event in local storage (file)

This PR also:
- removes logic of `QueueManager` as discussed internally p1699018266041689-slack-C0533SEJ82H
- makes save to file operation thread-safe of the main thread

### How to test
1. Run the test app
2. Tap on "Track URL" multiple times. Count how many times you tapped.
3. Observe logs: you'll notice a new log `Persisting n events`, where `n` is the number of events added in 1-second time frame
4. Move app to the background
5. Assert that number `m` of events in ` Sending request with m events` log equals the number you counted in step 2
